docs/development/extensions-contrib/k8s-jobs.md

```diff
@@ -286,4 +286,5 @@ To do this, set the following property.
 |--------|-----------------|-----------|-------|--------|
 |`druid.indexer.runner.k8sAndWorker.workerTaskRunnerType`|`String`|Determines whether the `httpRemote` or the `remote` task runner should be used in addition to the Kubernetes task runner.|`httpRemote`|No|
 |`druid.indexer.runner.k8sAndWorker.sendAllTasksToWorkerTaskRunner`|`boolean`| Whether to send all the tasks to the worker task runner. If this is set to false all tasks will be sent to Kubernetes|`false`|No|
+|`druid.indexer.runner.k8sAndWorker.runnerSelectorSpec`|`{"default":"k8s","overrides": {"index_kafka": "worker"}}`| Determines for a task type whether it should run in `k8s` or `worker` runner type|`false`|No|
```

Comment:
I think we should get rid of the first 2 configs since they are not yet released an instead push all this logic into the "runnerSelector". I imagine there could be 3 types of runner selectors, each with their own config. Something like how `druid.storage.type` is configured. We can preserve the existing default behavior ie if you do not select a runnerSelector, it defaults to the k8s selector. If you select a worker selector, it defaults to httpRemote runner.

I don't have a strong opinion on names yet, so if you think of a different naming scheme, that's fine with me.
```
# If someone wants to configure the k8s runner selector
druid.indexer.runner.k8sAndWorker.type=k8s (default)

# If someone wants to configure the worker runner selector
druid.indexer.runner.k8sAndWorker.type=worker
druid.indexer.runner.k8sAndWorker.useHttp=true (by default use httpRemote)

# If someone wants to configure the taskType runner selector
druid.indexer.runner.k8sAndWorker.type=taskType
druid.indexer.runner.k8sAndWorker.default=k8s (default)
druid.indexer.runner.k8sAndWorker.overrides={...}
```


druid.storage.type as an example (@JsonSubTypes)

Reply:
Updated with the new approach.